### PR TITLE
New interface for components that execute when the server is starting

### PR DESCRIPTION
**Pull request by slewis74**
This is used by the AD extension to spin up a 2nd WebHost, which can have a different AuthenticationScheme to the main site.

This PR also drops net452 for this library.

### CODE REVIEW

File: source/Server.Extensibility/Extensions/Infrastructure/Web/IExecuteWhenServerStarts.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+namespace Octopus.Server.Extensibility.Extensions.Infrastructure.Web
+{
+    public interface IExecuteWhenServerStarts
```
Comment (michaelnoonan) [inline source/Server.Extensibility/Extensions/Infrastructure/Web/IExecuteWhenServerStarts.cs]:
```suggestion
    public interface IExecuteWhenWebServerStarts
```

File: source/Server.Extensibility/Extensions/Infrastructure/Web/IExecuteWhenServerStarts.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    {
+        /// <summary>
+        /// Will be called when the web host is about to start
+        /// </summary>
+        void OnHostStarting();
```
Comment (michaelnoonan) [inline source/Server.Extensibility/Extensions/Infrastructure/Web/IExecuteWhenServerStarts.cs]:
```suggestion
        void OnWebHostStarting();
```

File: source/Server.Extensibility/Extensions/Infrastructure/Web/IExecuteWhenServerStarts.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+        /// <summary>
+        /// Will be called when the web host has started
+        /// </summary>
+        void OnHostStarted();
```
Comment (michaelnoonan) [inline source/Server.Extensibility/Extensions/Infrastructure/Web/IExecuteWhenServerStarts.cs]:
```suggestion
        void OnWebHostStarted();
```

File: source/Server.Extensibility/Extensions/Infrastructure/Web/IExecuteWhenWebHostStarts.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+using System.Threading.Tasks;
+
+namespace Octopus.Server.Extensibility.Extensions.Infrastructure.Web
+{
+    public interface IExecuteWhenWebHostStarts
```
Comment (johnsimons) [inline source/Server.Extensibility/Extensions/Infrastructure/Web/IExecuteWhenWebHostStarts.cs]: Could we line up this interface with something like https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.ihostedservice?view=dotnet-plat-ext-3.1

So what I am saying is that we rename it to `IWebHostService` and only include 2 methods:
`Task Start()` and `Task Stop()`

I honestly don't think we need the **ing** and **ed**.
In the implementation (which as far as I know is the only one) [we don't use `OnWebHostStarting`](https://github.com/OctopusDeploy/DirectoryServicesAuthenticationProvider/commit/d852aa2a026bf6b9818359100bc907f2eaa3dd05#diff-114e500eacbbdf6e0bd3edfa1eaf6c4dR36). 

I vote to keep it simple and line up with asp.net core already existing interfaces so when we actually go full asp.net core, the migration is simpler.

**Pull request merged**